File: java-backend/src/main/java/com/study/config/MinioProperties.java
package com.study.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * MinIO 配置,绑定 minio.* 。
 *
 * @ConfigurationProperties 把 minio.endpoint/access-key/secret-key/bucket 绑定到这个对象。
 */
@ConfigurationProperties(prefix = "minio")
public class MinioProperties {

    private String endpoint;
    private String accessKey;
    private String secretKey;
    private String bucket;

    public String getEndpoint() {
        return endpoint;
    }

    public void setEndpoint(String endpoint) {
        this.endpoint = endpoint;
    }

    public String getAccessKey() {
        return accessKey;
    }

    public void setAccessKey(String accessKey) {
        this.accessKey = accessKey;
    }

    public String getSecretKey() {
        return secretKey;
    }

    public void setSecretKey(String secretKey) {
        this.secretKey = secretKey;
    }

    public String getBucket() {
        return bucket;
    }

    public void setBucket(String bucket) {
        this.bucket = bucket;
    }
}

File: kotlin-backend/src/main/kotlin/com/study/config/MinioProperties.kt
package com.study.config

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * MinIO 配置(构造器绑定):minio.* 。
 *
 * @ConfigurationProperties 把 minio.endpoint/access-key/secret-key/bucket 绑定到这个 data class。
 */
@ConfigurationProperties(prefix = "minio")
data class MinioProperties(
    val endpoint: String,
    val accessKey: String,
    val secretKey: String,
    val bucket: String,
)
